CMakeLists.txt

```diff
@@ -492,45 +540,100 @@ if(QUDA_MAGMA)
+                        CMAKE_ARGS -DMPI=${ARPACK_MPI} -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+                        CMAKE_GENERATOR "Unix Makefiles"
+                        # BUILD_COMMAND  make
+                        # INSTALL_COMMAND make install
+                        # LOG_INSTALL ON
+                        # LOG_BUILD ON
+                        # LOG_DOWNLOAD ON
+                        # LOG_MERGED_STDOUTERR ON
+                        # LOG_OUTPUT_ON_FAILURE ON
+                        )
+    ExternalProject_Get_Property(ARPACK-NG INSTALL_DIR)
+    set(QUDA_ARPACK_HOME ${INSTALL_DIR})
+    add_library(arpack-ng STATIC IMPORTED)
+    add_dependencies(arpack-ng ARPACK-NG)
+    find_package(BLAS REQUIRED)
+    find_package(LAPACK REQUIRED)
+    # target_link_libraries(arpack-ng INTERFACE  -L${QUDA_ARPACK_HOME}/lib -larpack)
+    target_link_libraries(arpack-ng INTERFACE ${BLAS_LIBRARIES} ${LAPACK_LIBRARIES})
+    set_target_properties(arpack-ng PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES Fortran)
+    set_target_properties(arpack-ng
+                          PROPERTIES IMPORTED_LOCATION ${QUDA_ARPACK_HOME}/${CMAKE_INSTALL_LIBDIR}/libarpack.a)
+    if(QUDA_MPI OR QUDA_QMP)
+      add_library(parpack-ng STATIC IMPORTED)
+      # target_link_libraries(arpack-ng INTERFACE -L${QUDA_ARPACK_HOME}/lib -lparpack)
+      target_link_libraries(parpack-ng INTERFACE arpack-ng MPI::MPI_Fortran)
+      set_target_properties(parpack-ng PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES Fortran)
+      set_target_properties(parpack-ng
+                            PROPERTIES IMPORTED_LOCATION ${QUDA_ARPACK_HOME}/${CMAKE_INSTALL_LIBDIR}/libparpack.a)
+      # include_directories(SYSTEM ${QUDA_ARPACK_HOME}/PARPACK/SRC/MPI) include_directories(SYSTEM
+      # ${QUDA_ARPACK_HOME}/PARPACK/UTIL/MPI)
+    endif()
+
+    # ARPACK-NG does not seem to suppport logging - we need to disable it
+    add_definitions(-DARPACK_NG)
+  else(QUDA_DOWNLOAD_ARPACK)
+    find_package(PkgConfig REQUIRED)
+
+    # We always need the serial library
+    pkg_check_modules(ARPACK QUIET arpack)
+    if(NOT ARPACK_FOUND OR QUDA_ARPACK_HOME)
+      find_library(ARPACK arpack PATH ${QUDA_ARPACK_HOME})
     else()
-      find_library(PARPACK ${PARPACK_LIBRARIES} PATH ${PARPACK_LIBRARY_DIRS})
+      find_library(ARPACK ${ARPACK_LIBRARIES} PATH ${ARPACK_LIBRARY_DIRS})
+    endif()
+   
+    # Link the parallel library if required
+    if(QUDA_MPI OR QUDA_QMP)
+      pkg_check_modules(PARPACK QUIET parpack)
+      if(NOT PARPACK_FOUND OR QUDA_ARPACK_HOME)
+        find_library(PARPACK parpack PATH ${QUDA_ARPACK_HOME})
+      else()
+        find_library(PARPACK ${PARPACK_LIBRARIES} PATH ${PARPACK_LIBRARY_DIRS})
+      endif()
     endif()
-    include_directories(SYSTEM ${QUDA_ARPACK_HOME}/PARPACK/SRC/MPI)
-    include_directories(SYSTEM ${QUDA_ARPACK_HOME}/PARPACK/UTIL/MPI)
+  endif(QUDA_DOWNLOAD_ARPACK)
+
+   include_directories(SYSTEM ${QUDA_ARPACK_HOME}/SRC)
```

Comment:
@cpviolator Do you think we need these include to `SRC` and `UTIL`? All I get from them are warnings. And removing them does not break compilation for me.